src/hooks/useLiveLogs.ts

```
@@ -0,0 +1,406 @@
+      message: generateDataFetchingMessage(activityType, data),
+      timestamp: (data.timestamp ? data.timestamp * 1000 : Date.now()),
+      data: data,
+      toolType: 'data_tool'
+    };
+    
+    setLiveLogEntries(prev => [logEntry, ...prev].slice(0, 100));
+  };
+  
+  // Handler for reasoning_started events
+  const handleReasoningStarted = (activityType: string, data: any) => {
+    const reasoningId = generateReasoningId();
```

ESLint

[error] 'generateReasoningId' was used before it was defined.

(@typescript-eslint/no-use-before-define)